<a href="https://colab.research.google.com/github/tastiz/SQL-skills/blob/master/ksprojects.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
from sqlalchemy import create_engine

postgres_user = 'dabc_student'
postgres_pw = '7*.8G9QH21'
postgres_host = '142.93.121.174'
postgres_port = '5432'
postgres_db = 'kickstarterprojects'

engine = create_engine('postgresql://{}:{}@{}:{}/{}'.format(
    postgres_user, postgres_pw, postgres_host, postgres_port, postgres_db))

import pandas as pd

sql = 'SELECT * FROM ksprojects '

kickstarter_df = pd.read_sql_query(sql, con=engine)

kickstarter_df.shape

kickstarter_df.info()

#kickstarter_df.describe()

sql2 = 'SELECT column_names FROM ksprojects WHERE column_name IS NULL'

null_values = pd.read_sql_query(sql2, con=engine)

null_values.head()



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65418 entries, 0 to 65417
Data columns (total 13 columns):
id               65418 non-null int64
name             65418 non-null object
category         65418 non-null object
main_category    65418 non-null object
deadline         65418 non-null datetime64[ns]
goal             65418 non-null float64
launched         65418 non-null datetime64[ns]
state            65418 non-null object
backers          65418 non-null int64
country          65418 non-null object
usd_pledged      64673 non-null float64
currency         65418 non-null object
pledged          65418 non-null float64
dtypes: datetime64[ns](2), float64(3), int64(2), object(6)
memory usage: 6.5+ MB


ProgrammingError: ignored

There are 65418 rows and 13 columns in the ksprojects table



id        :         int64
name       :        object
category     :      object
main_category :     object
deadline       :    datetime64
goal            :   float64
launched        :   datetime64
state           :   object
backers         :   int64
country         :   object
usd_pledged     :  float64
currency           object
pledged        :    float64